python/lsst/ts/ess/csc/rpi_data_client.py

```diff
@@ -34,16 +34,19 @@
 from lsst.ts import salobj, tcpip
 from lsst.ts.ess import common
 
-# Time limit for connecting to the RPi (seconds)
+# Time limit for connecting to the RPi (seconds).
 CONNECT_TIMEOUT = 5
 
-# Time limit for communicating with the RPi (seconds)
-# This includes writing a command and reading the response
-# and reading telemetry (seconds)
-COMMUNICATE_TIMEOUT = 5
+# The number of timeouts to allow before raising a TimeoutError.
+NUM_ALLOWED_TIMEOUTS = 5
 
 
 class RPiDataClient(common.BaseDataClient):
+    # Timeout limit for communicating with the RPi (seconds). This includes
+    # writing a command and reading the response and reading telemetry. Unit
+    # tests can set this to a lower value to speed up the test.
+    communicate_timeout = 60
```

Comment:
Is it worth using this same timeout for connection? If the network is so slow that we need 60 seconds to communicate then we might also need that much time to initiate the connection. Though that will sure slow down transitioning to enabled state, so maybe it's better to live with it as it is.

Reply:
Sounds great.